### PR TITLE
Added AbstractCursorItemReader.connectionAutoCommit

### PR DESCRIPTION
**Pull request by gdarmont**
This PR is a follow up of discussion in PR https://github.com/spring-projects/spring-boot/pull/7866

The use case was to use a JdbcCursorItemReader with PostgreSQL (nothing fancy here).
From https://jdbc.postgresql.org/documentation/94/query.html, section "Getting results based on a cursor", the connection must not be in auto commit mode.
But JdbcCursorItemReader does not allow to explicity set connection mode to autoCommit = false. 

As discussed in the Spring Boot PR, configuring the JDBC pool to force all connection to autoCommit=false is not an optimal way to workaround the problem.

So here's a little PR that allows configuration of autocommit mode for the connection used by a CursorItemReader.

I validated the use of a Boolean wrapper against https://github.com/spring-projects/spring-framework/blob/master/spring-jdbc/src/main/java/org/springframework/jdbc/datasource/SingleConnectionDataSource.java#L65, to keep Spring philosophy.

Hope this helps,
Guillaume

### CODE REVIEW

File: spring-batch-infrastructure/src/main/java/org/springframework/batch/item/database/AbstractCursorItemReader.java [hunk 1]
```diff
@@ -424,6 +444,9 @@ protected void initializeConnection() {
 			else {
 				this.con = dataSource.getConnection();
 			}
+			if (this.connectionAutoCommit != null && this.con.getAutoCommit() != this.connectionAutoCommit) {
+				this.con.setAutoCommit(this.connectionAutoCommit);
+			}
```
Comment (mminella) [inline spring-batch-infrastructure/src/main/java/org/springframework/batch/item/database/AbstractCursorItemReader.java]: If we allow this, we need to set it back to what it was once we're done with it.

Comment (gdarmont) [inline spring-batch-infrastructure/src/main/java/org/springframework/batch/item/database/AbstractCursorItemReader.java]: Thanks for the review.
I added the necessary code and amended the commit.

**Comment by gdarmont** [comment]
Hi Spring Batch Team,

Do you think there is additional work to to on this PR ? I'd be happy to help.

Guillaume

**Comment by mminella** [comment]
Rebased and merged as dca726af16b7145bb2950d1709c8eaaf77ede7a4.  Also created Jira issue BATCH-2661 for this.  Thanks!

**Pull request closed without merging**